File: problems/reorder-list/stack.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    // Using a stack as a solution
    public void ReorderList(ListNode head) {
        var length = 0;
        var current = head;
        var stack = new Stack<ListNode>();
        while (current is not null)
        {
            length++;
            stack.Push(current);
            current = current.next;
        }

        Console.WriteLine(length);

        current = head;
        for (var i = 0; i < length; i++)
        {
            var original = current;
            if (i % 2 == 0)
            {
                var n = stack.Pop();
                n.next = original.next;
                original.next = n;
            }
            current = current.next;
        }

        current.next = null;
    }
}

File: problems/remove-all-adjacent-duplicates-in-string/stack.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []

        for letter in s:
            if len(stack) > 0 and stack[-1] == letter:
                stack.pop()
            else:
                stack.append(letter)

        print(stack)
        output = ""
        for letter in stack:
            output += letter

        return output